ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -365,27 +364,38 @@ def _clear_instance_recordings(self, instance: Any) -> None:
         if hasattr(instance, "_datadog_spans"):
             delattr(instance, "_datadog_spans")
 
-    def _llmobs_set_metadata(self, span: Span, model_provider: Optional[str] = None) -> None:
-        if not model_provider:
+    def _llmobs_set_metadata(self, span: Span, kwargs: Dict[str, Any]) -> None:
+        identifying_params = kwargs.pop("_dd.identifying_params", None)
+        if not identifying_params:
             return
+        metadata: Dict[str, Any] = {}
+        for _, val in identifying_params.items():
+            if isinstance(val, dict):
+                params = self._llmobs_extract_parameters(val)
+            else:
+                params = self._llmobs_extract_parameters(identifying_params)
```

Comment:
```suggestion
        metadata = self._llmobs_extract_parameters(identifying_params)
        for v in identifying_params.values():
            if not isinstance(v, dict):
                continue
            metadata = self._llmobs_extract_parameters(val)
            if metadata:
                break
```
I'm thinking we just find the first set of max_tokens/temperature from the identifying_params variable. No need to try the same thing for every single kv pair